templates/_layouts/docs.html

```diff
@@ -48,145 +48,25 @@
-                {{ side_nav_item("/docs/patterns/pagination", "Pagination") }}
-                {{ side_nav_item("/docs/patterns/pull-quote", "Quotes") }}
-                {{ side_nav_item("/docs/patterns/search-and-filter", "Search and filter") }}
-                {{ side_nav_item("/docs/patterns/search-box", "Search box") }}
-                {{ side_nav_item("/docs/patterns/segmented-control", "Segmented control") }}
-                {{ side_nav_item("/docs/patterns/slider", "Slider") }}
-                {{ side_nav_item("/docs/patterns/status-labels", "Status labels") }}
-                {{ side_nav_item("/docs/patterns/strip", "Strip") }}
-                {{ side_nav_item("/docs/patterns/switch", "Switch") }}
-                {{ side_nav_item("/docs/patterns/table-of-contents", "Table of contents") }}
-                {{ side_nav_item("/docs/patterns/tabs", "Tabs") }}
-                {{ side_nav_item("/docs/patterns/tooltips", "Tooltips") }}
-              </ul>
-
-              <ul class="p-side-navigation__list">
-                <li class="p-side-navigation__item--title"><span class="p-side-navigation__text">Utilities</span></li>
-                {{ side_nav_item("/docs/utilities/align", "Align") }}
-                {{ side_nav_item("/docs/utilities/baseline-grid", "Baseline grid") }}
-                {{ side_nav_item("/docs/utilities/clearfix", "Clearfix") }}
-                {{ side_nav_item("/docs/utilities/embedded-media", "Embedded media") }}
-                {{ side_nav_item("/docs/utilities/equal-height", "Equal height") }}
-                {{ side_nav_item("/docs/utilities/floats", "Floats") }}
-                {{ side_nav_item("/docs/utilities/font-metrics", "Font metrics") }}
-                {{ side_nav_item("/docs/utilities/functions", "Functions") }}
-                {{ side_nav_item("/docs/utilities/hide", "Hide") }}
-                {{ side_nav_item("/docs/utilities/image-position", "Image position") }}
-                {{ side_nav_item("/docs/utilities/margin-collapse", "Margin collapse") }}
-                {{ side_nav_item("/docs/utilities/no-print", "No print") }}
-                {{ side_nav_item("/docs/utilities/off-screen", "Off-screen") }}
-                {{ side_nav_item("/docs/utilities/padding-collapse", "Padding collapse") }}
-                {{ side_nav_item("/docs/utilities/show", "Show") }}
-                {{ side_nav_item("/docs/utilities/table-cell-padding-overlap", "Table cell padding overlap") }}
-                {{ side_nav_item("/docs/utilities/has-icon", "Text with icon") }}
-                {{ side_nav_item("/docs/utilities/truncate", "Truncation") }}
-                {{ side_nav_item("/docs/utilities/vertical-spacing", "Vertical spacing") }}
-                {{ side_nav_item("/docs/utilities/vertically-center", "Vertically center") }}
-              </ul>
-
+              
+              {% for title in sideNavigation %}
```

Comment:
The `title` name is a bit misleading here, because it has `heading` and `subheadings`, I think `item in sideNavigation` or `section in sideNavigation` will make the naming more natural.